src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -87,23 +91,47 @@ void DeleteInteractionChoiceSetRequest::Run() {
     return;
   }
   SendVrDeleteCommand(app);
+}
 
-  smart_objects::SmartObject msg_params =
-      smart_objects::SmartObject(smart_objects::SmartType_Map);
+bool DeleteInteractionChoiceSetRequest::Init() {
+  hash_update_mode_ = HashUpdateMode::kDoHashUpdate;
+  return true;
+}
 
-  msg_params[strings::interaction_choice_set_id] = choice_set_id;
-  msg_params[strings::app_id] = app->app_id();
+void DeleteInteractionChoiceSetRequest::on_event(
+    const event_engine::Event& event) {
+  using namespace helpers;
+  LOG4CXX_AUTO_TRACE(logger_);
 
-  app->RemoveChoiceSet(choice_set_id);
+  if (event.id() == hmi_apis::FunctionID::VR_DeleteCommand) {
+    {
+      sync_primitives::AutoLock auto_lock(requests_lock_);
+      const smart_objects::SmartObject& message = event.smart_object();
+      response_result_code_ = static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asInt());
+      const std::uint32_t correlation_id = static_cast<uint32_t>(
+          message[strings::params][strings::correlation_id].asUInt());
+
+      auto found_request = std::find(
```

Comment:
@IGapchuk use `sent_requests_.find(correlation_id)`

Reply:
@BSolonenko Changed in 05a0ac563f1b887f483b99518f63582d86643583.